lib/class-wp-duotone-gutenberg.php

```diff
@@ -360,11 +360,11 @@ public static function render_duotone_support( $block_content, $block ) {
 		);
 
 		// Like the layout hook, this assumes the hook only applies to blocks with a single wrapper.
-		return preg_replace(
-			'/' . preg_quote( 'class="', '/' ) . '/',
-			'class="' . $filter_id . ' ',
-			$block_content,
-			1
-		);
+		$tags = new WP_HTML_Tag_Processor( $block_content );
+		if ( $tags->next_tag() ) {
+			$tags->add_class( $filter_id );
+		}
```

Comment:
It looks like [`add_class` gets a free pass](https://github.com/WordPress/gutenberg/blob/0ab1a64d2dadd663099cf659342580fe14bc0c71/lib/experimental/html/class-wp-html-tag-processor.php#L130-L145) and the if statement isn't necessary/desired here.

```suggestion
		$tags->add_class( $filter_id );
```

Reply:
The `$tags->next_tag()` returns a boolean if it finds a tag or not, so the `if` statement is checking to make sure that we have an element to add the class to, not that the class already exists on the element. The only reason I think it might return `false` is if you have an empty string or malformed HTML.

For reference, I based this off of how it was used in https://github.com/WordPress/gutenberg/pull/46625.